test/check/check.go

```diff
@@ -0,0 +1,270 @@
+		log.Fail("Condition %q has not reached desired state %q within given timeout", conditionType, conditionStatus)
+		return err
+	}
+	log.Ok("Condition %q has reached desired state %q", conditionType, conditionStatus)
+	return nil
+}
+
+func checkForComponent(component *Component) error {
+	errs := []error{}
+	errsAppend := func(err error) {
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	if component.Namespace != "" {
+		errsAppend(checkForNamespace(component.Namespace))
+	}
+
+	if component.ClusterRole != "" {
+		errsAppend(checkForClusterRole(component.ClusterRole))
+	}
+
+	if component.ClusterRoleBinding != "" {
+		errsAppend(checkForClusterRoleBinding(component.ClusterRoleBinding))
+	}
+
+	if component.SecurityContextConstraints != "" {
+		errsAppend(checkForSecurityContextConstraints(component.SecurityContextConstraints))
+	}
+
+	for _, daemonSet := range component.DaemonSets {
+		errsAppend(checkForDaemonSet(daemonSet, component.Namespace))
+	}
+
+	for _, deployment := range component.Deployments {
+		errsAppend(checkForDeployment(deployment, component.Namespace))
+	}
+
+	return errsToErr(errs)
+}
+
+func waitForComponentRemoval(component *Component) error {
+	errs := []error{}
+	errsAppend := func(err error) {
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	if component.Namespace != "" {
+		errsAppend(waitForNamespaceRemoval(component.Namespace, 4*time.Minute))
+	}
+
+	if component.ClusterRole != "" {
+		errsAppend(waitForClusterRoleRemoval(component.ClusterRole, 10*time.Second))
+	}
+
+	if component.ClusterRoleBinding != "" {
+		errsAppend(waitForClusterRoleBindingRemoval(component.ClusterRoleBinding, 10*time.Second))
+	}
+
+	if component.SecurityContextConstraints != "" {
+		errsAppend(waitForSecurityContextConstraintsRemoval(component.SecurityContextConstraints, 10*time.Second))
+	}
+
+	return errsToErr(errs)
+}
+
+func errsToErr(errs []error) error {
+	if len(errs) == 0 {
+		return nil
+	}
+	errsStrings := []string{}
+	for _, err := range errs {
+		errsStrings = append(errsStrings, err.Error())
+	}
+	return errors.New(strings.Join(errsStrings, "\n"))
+}
+
+func checkForNamespace(name string) error {
+	return framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: name}, &corev1.Namespace{})
+}
+
+func checkForClusterRole(name string) error {
+	return framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: name}, &rbacv1.ClusterRole{})
+}
+
+func checkForClusterRoleBinding(name string) error {
+	return framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: name}, &rbacv1.ClusterRoleBinding{})
+}
+
+func checkForSecurityContextConstraints(name string) error {
+	err := framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: name}, &securityapi.SecurityContextConstraints{})
+	if strings.Contains(err.Error(), "no kind is registered for the type") {
+		return nil
+	}
+	return err
+}
+
+func checkForDeployment(name string, namespace string) error {
```

Comment:
Can we use operator-sdk [waitForDeployment](https://github.com/operator-framework/operator-sdk/blob/master/pkg/test/e2eutil/wait_util.go#L46) ?

Reply:
I don't want to poll. Check calls is to be called after Ready condition turns True. At that moment, all components should be available. If they aren't, it is a bug in our condition handling.